forge-gui/res/cardsfolder/upcoming/modify_memory.txt

```diff
@@ -0,0 +1,7 @@
+Name:Modify Memory
+ManaCost:4 U
+Types:Sorcery
+A:SP$ ExchangeControl | ValidTgts$ Creature | TgtPrompt$ Select target creature | TargetMin$ 2 | TargetMax$ 2 | TargetsWithDifferentControllers$ True | RememberExchanged$ True | SubAbility$ DBDraw | SpellDescription$ Exchange control of two target creatures controlled by different players.
+SVar:DBDraw:DB$ Draw | NumCards$ 3 | ConditionDefined$ Remembered | ConditionPresent$ Creature.YouCtrl | ConditionCompare$ EQ0 | SubAbility$ DBCleanup | SubAbility$ DBCleanup | StackDescription$ SpellDescription | SpellDescription$ If you control neither creature, draw three cards.
```

Comment:
This one is actually a bit complicated rules-wise, see here:
https://www.reddit.com/r/mtgrules/comments/v9399j/modify_memory_yes_again/

> You don't get to draw cards if one target is illegal, no matter who controlled it before the spell resolved.

Reply:
I think extend ControlExchangeEffect class resolve function to have a param like "ReplaceFailedExchange$ ReplacementEffect" or just a flag if the exchange failed do something else..

Reply:
Btw, this line seems to have two references to SubAbility$ DBCleanup, probably just a duplicate?